app/packages/core/src/components/ColorModal/LabelTag.tsx

```diff
@@ -0,0 +1,235 @@
+    setShowFieldPicker(false);
+  });
+
+  return (
+    <div>
+      <ModeControl />
+      <Divider />
+
+      {coloring.by === "field" && (
+        <div style={{ margin: "1rem", width: "100%" }}>
+          <Checkbox
+            name={`Use custom color for label tags field`}
+            value={state.useFieldColor}
+            setValue={(v: boolean) => {
+              const copy = cloneDeep(labelTags);
+              copy.fieldColor = v ? labelTags?.fieldColor : undefined;
+              setColorScheme(false, {
+                colorPool,
+                fields,
+                labelTags: copy,
+              });
+              setState((s) => ({ ...s, useFieldColor: v }));
+            }}
+          />
+          {state?.useFieldColor && (
+            <div
+              style={{
+                margin: "1rem",
+                display: "flex",
+                flexDirection: "row",
+                alignItems: "end",
+              }}
+            >
+              <FieldColorSquare
+                color={labelTags?.fieldColor ?? input}
+                onClick={toggleColorPicker}
+                id="color-square"
+              >
+                {showFieldPicker && (
+                  <PickerWrapper
+                    id="twitter-color-container"
+                    onBlur={hideFieldColorPicker}
+                    visible={showFieldPicker}
+                    tabIndex={0}
+                    ref={wrapperRef}
+                  >
+                    <TwitterPicker
+                      color={input ?? labelTags?.fieldColor}
+                      colors={colors}
+                      onChange={(color) => setInput(color.hex)}
+                      onChangeComplete={(color) => {
+                        onChangeFieldColor(color.hex);
+                        setColors([...new Set([...colors, color.hex])]);
+                      }}
+                      className={colorPicker}
+                      ref={pickerRef}
+                    />
+                  </PickerWrapper>
+                )}
+              </FieldColorSquare>
+              <Input
+                value={input}
+                setter={(v) => setInput(v)}
+                onBlur={() => onValidateColor(input)}
+                onEnter={() => onValidateColor(input)}
+                style={{
+                  width: 120,
+                  display: "inline-block",
+                  margin: 3,
+                }}
+              />
+            </div>
+          )}
+        </div>
+      )}
+
+      {coloring.by === "value" && (
+        <div>
+          <form
+            style={{ display: "flex", flexDirection: "column", margin: "1rem" }}
+          >
+            {/* set attribute value - color */}
+            <Checkbox
+              name={`Use custom colors for specific field values`}
+              value={state.useLabelColors}
+              setValue={(v: boolean) => {
+                const copy = cloneDeep(labelTags);
```

Comment:
I did see the cloneDeep multiple times. Maybe you can define it earlier one and reuse?